src/main/java/pl/kl/weatherservice/location/LocationsService.java

```diff
@@ -0,0 +1,66 @@
+package pl.kl.weatherservice.location;
+
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+import pl.kl.weatherservice.exceptions.EmptyInputException;
+import pl.kl.weatherservice.exceptions.InputOutOfRangeException;
+
+import java.util.Optional;
+import java.util.function.Predicate;
+
+@Component
+@RequiredArgsConstructor
+class LocationsService {
+
+    private final Double NORTH_LIMIT = 90.0;
+    private final Double SOUTH_LIMIT = -90.0;
+    private final Double EAST_LIMIT = 180.0;
+    private final Double WEST_LIMIT = -180.0;
+
+    private final LocationsRepository locationsRepository;
+
+    Location createLocation(String city, String region, String country, Double latitude, Double longitude) {
+        validateCity(city);
+        validateCountry(country);
+        validateLatitude(latitude);
+        validateLongitude(longitude);
+
+        Location location = new Location();
+        location.setCity(city);
+        location.setRegion(Optional.ofNullable(region).filter(Predicate.not(String::isBlank)).orElse(null));
```

Comment:
`Optional.ofNullable(region).filter(Predicate.not(String::isBlank)).ifPresent(location::setRegion)`

Reply:
Fixed